File: src/main/java/org/acme/Movie.java
package org.acme;

import io.quarkus.mongodb.panache.PanacheMongoEntity;
import io.quarkus.mongodb.panache.common.MongoEntity;
import org.bson.types.ObjectId;

@MongoEntity(collection = "movies")
public class Movie extends PanacheMongoEntity {

    public String title;
    public String category;
    public int duration;

    public Movie() {
    }

    public Movie(String title, String category, int duration) {
        this.title = title;
        this.category = category;
        this.duration = duration;
    }
}

File: frontend/src/types/Movie.ts
export interface Movie {
  id: string | { $oid: string };
  title: string;
  category: string;
  duration: number;
}

export interface MovieCreate {
  title: string;
  category: string;
  duration: number;
}

export function getMovieId(movie: Movie): string {
  if (typeof movie.id === 'string') return movie.id;
  return movie.id.$oid;
}
